src/app/system/manage-hooks/create-hook/create-hook.component.ts

```diff
@@ -0,0 +1,176 @@
+
+/**
+ * Create Hook Component.
+ */
+@Component({
+  selector: 'mifosx-create-hook',
+  templateUrl: './create-hook.component.html',
+  styleUrls: ['./create-hook.component.scss']
+})
+export class CreateHookComponent implements OnInit {
+
+  /** Hooks Template Data. */
+  hooksTemplateData: any;
+  /** Hook Form. */
+  hookForm: FormGroup;
+  /** Entity Data. */
+  entityData: Array<any> = new Array<any>();
+  /** Action Data. */
+  actionData: Array<any> = new Array<any>();
+
+  /**
+   * Retrieves the hooks template data from `resolve`.
+   * @param {ActivatedRoute} route Activated Route.
+   * @param {SystemService} systemService System Service.
+   * @param {Router} router Router for navigation.
+   * @param {FormBuilder} formBuilder Form Builder.
+   */
+  constructor(private route: ActivatedRoute,
+              private systemService: SystemService,
+              private router: Router,
+              private formBuilder: FormBuilder) {
+    this.route.data.subscribe((data: { hooksTemplate: any }) => {
+      this.hooksTemplateData = data.hooksTemplate;
+    });
+  }
+
+  /**
+   * Creates the hook form.
+   */
+  ngOnInit() {
+    this.createHookForm();
+    this.hookForm.get('name').valueChanges.subscribe(name => {
+      if (name === 'Web') {
+        this.hookForm.get('contentType').enable();
+        this.hookForm.get('phoneNumber').disable();
+        this.hookForm.get('smsProvider').disable();
+        this.hookForm.get('smsProviderAccountId').disable();
+        this.hookForm.get('smsProviderToken').disable();
+      } else {
+        this.hookForm.get('contentType').disable();
+        this.hookForm.get('phoneNumber').enable();
+        this.hookForm.get('smsProvider').enable();
+        this.hookForm.get('smsProviderAccountId').enable();
+        this.hookForm.get('smsProviderToken').enable();
+      }
+    });
+  }
+
+  /**
+   * Creates the hook form.
+   */
+  createHookForm() {
+    this.hookForm = this.formBuilder.group({
+      'name': ['Web', Validators.required],
+      'displayName': ['', Validators.required],
+      'isActive': [''],
+      'phoneNumber': [{ value: '', disabled: true }, Validators.required],
+      'smsProvider': [{ value: '', disabled: true }, Validators.required],
+      'smsProviderAccountId': [{ value: '', disabled: true }, Validators.required],
+      'smsProviderToken': [{ value: '', disabled: true }, Validators.required],
+      'contentType': ['', Validators.required],
+      'payloadUrl': ['', Validators.required],
+      'events': this.formBuilder.array([this.createEventForm()])
+    });
+    this.setGroupingListener(0);
+    this.setEntityListener(0);
+  }
+
+  /**
+   * Creates the Event Form.
+   * @returns {FormGroup} Event Form.
+   */
+  createEventForm(): FormGroup {
+    return this.formBuilder.group({
+      'grouping': ['', Validators.required],
+      'entityName': ['', Validators.required],
+      'actionName': ['', Validators.required]
+    });
+  }
+
```

Comment:
Is it necessary? I thought it was an improved version of community-app

Reply:
No, its not necessary but the current implementation is not user friendly in my opinion instead the previous implementation was better if you can improve your UI then good.

The reason I feel its not user friendly are:

1. Add button is beside the fields in which the user has to provide input not indicating at first glance whether the user has to press that button or not, even if he wants to add only one param or when he wants to add more than one param.

2. When we edit the delete button is not visible in case of single param. till the time user press on add button. this could be improved by showing the delete button and when the user press the delete button an alert window opens and inform the user to add another param before he can delete previous param.

so there is lot of functionality that needs to be discovered before by the user himself.

Reply:
Your first point is not resolved in the case of old implementation as the table is not shown until you press the add button.

Sorry, I am not able to understand what are you trying to explain in the second point.

Reply:
that's the point table element which contains all the entries that are added to the hook. So if its not visiable it means that no route params are added to hook 

where as in the your implementation both the element where data is entered by the user is the same element which shows the final list of params added by the user to the hook so what i want to convey is there should be clear design cue(hints) to the user what button is for what action.

Reply:
If that is the case, let us get the view of a user, @santoshconflux can you please help?

Just wanted to add a point, if you don't press the add button and have selected all the three fields it will get passed in the API request.

Reply:
On thinking about a your UI can be simply improved by just moving add button beside Route Parameter heading. 
On pressing that button new fields become visible with there delete button beside them.

Reply:
> If that is the case, let us get the view of a user, @santoshconflux can you please help?
> 
> Just wanted to add a point, if you don't press the add button and have selected all the three fields it will get passed in the API request.

I know that i  am not saying your implementation is wrong in any way i was just suggesting a UI related suggestion.

Reply:
We need at least one of that.